security: [HIGH] Command injection via base64 shell interpolation in daytona.sh _exec_long

## Finding
**File**: sh/e2e/lib/clouds/daytona.sh:236
**Severity**: HIGH

**Description**: The `_daytona_exec_long` function base64-encodes commands but interpolates the encoded value directly into a double-quoted shell string, creating potential for command injection.

```bash
# Line 232-236
local encoded_cmd
encoded_cmd=$(printf '%s' "${cmd}" | base64 | tr -d '\n')

# shellcheck disable=SC2086
ssh ${ssh_args} "${_DT_TOKEN}@${_DT_HOST}" "timeout ${timeout} bash -c \"\$(printf '%s' '${encoded_cmd}' | base64 -d)\""
```

The problem: `${encoded_cmd}` is interpolated into a double-quoted string. If the base64 output contains characters like `$`, backticks, or `\`, they will be interpreted by the local shell BEFORE being sent to SSH.

While base64 output is typically safe (alphanumeric + `+/=`), this pattern violates defense-in-depth and is fragile:
1. Different base64 implementations may produce different output
2. The `tr -d '\n'` could fail silently
3. Future modifications could break the safety assumption

**Similar Issue**: This is the same pattern that was flagged in #2227, #2071, and #2164 for other files.

**Recommendation**: 
Use the proven safe pattern from other cloud drivers (e.g., aws.sh, hetzner.sh):

```bash
# Pass via stdin instead of shell interpolation
printf '%s' "${cmd}" | ssh ${ssh_args} "${_DT_TOKEN}@${_DT_HOST}" \
  "timeout ${timeout} bash -c '\$(base64 -d)'```

Or use SSH's command execution with array arguments if the command doesn't need to be hidden.

---
*-- security/shell-scanner*

--- a/sh/e2e/lib/clouds/aws.sh
+++ b/sh/e2e/lib/clouds/aws.sh
@@ -174,14 +174,12 @@ _aws_exec_long() {
 
   local alive_count=$((timeout / 15 + 1))
 
-  # Base64-encode the command to avoid shell injection via single-quote breakout
-  local encoded_cmd
-  encoded_cmd=$(printf '%s' "${cmd}" | base64 | tr -d '\n')
-
-  ssh -o StrictHostKeyChecking=no -o UserKnownHostsFile=/dev/null \
+  # Pipe the command via stdin to avoid interpolating it into the remote
+  # command string — eliminates shell injection risk from base64 encoding.
+  printf '%s' "${cmd}" | ssh -o StrictHostKeyChecking=no -o UserKnownHostsFile=/dev/null \
       -o ConnectTimeout=10 -o LogLevel=ERROR -o BatchMode=yes \
       -o "ServerAliveInterval=15" -o "ServerAliveCountMax=${alive_count}" \
-      "ubuntu@${_AWS_INSTANCE_IP}" "timeout ${timeout} bash -c \"\$(printf '%s' '${encoded_cmd}' | base64 -d)\""
+      "ubuntu@${_AWS_INSTANCE_IP}" "timeout ${timeout} bash"
 }
 
 # ---------------------------------------------------------------------------

--- a/sh/e2e/lib/clouds/digitalocean.sh
+++ b/sh/e2e/lib/clouds/digitalocean.sh
@@ -179,14 +179,12 @@ _digitalocean_exec_long() {
     return 1
   fi
 
-  # Base64-encode the command to avoid shell injection via single-quote breakout
-  local encoded_cmd
-  encoded_cmd=$(printf '%s' "${cmd}" | base64 | tr -d '\n')
-
-  ssh -o StrictHostKeyChecking=no -o UserKnownHostsFile=/dev/null \
+  # Pipe the command via stdin to avoid interpolating it into the remote
+  # command string — eliminates shell injection risk from base64 encoding.
+  printf '%s' "${cmd}" | ssh -o StrictHostKeyChecking=no -o UserKnownHostsFile=/dev/null \
       -o ConnectTimeout=10 -o LogLevel=ERROR -o BatchMode=yes \
       -o "ServerAliveInterval=15" -o "ServerAliveCountMax=$((timeout_secs / 15 + 1))" \
-      "root@${ip}" "timeout ${timeout_secs} bash -c \"\$(printf '%s' '${encoded_cmd}' | base64 -d)\""
+      "root@${ip}" "timeout ${timeout_secs} bash"
 }
 
 # ---------------------------------------------------------------------------

--- a/sh/e2e/lib/clouds/gcp.sh
+++ b/sh/e2e/lib/clouds/gcp.sh
@@ -182,14 +182,12 @@ _gcp_exec_long() {
 
   local alive_count=$((timeout / 15 + 1))
 
-  # Base64-encode the command to avoid shell injection via single-quote breakout
-  local encoded_cmd
-  encoded_cmd=$(printf '%s' "${cmd}" | base64 | tr -d '\n')
-
-  ssh -o StrictHostKeyChecking=no -o UserKnownHostsFile=/dev/null \
+  # Pipe the command via stdin to avoid interpolating it into the remote
+  # command string — eliminates shell injection risk from base64 encoding.
+  printf '%s' "${cmd}" | ssh -o StrictHostKeyChecking=no -o UserKnownHostsFile=/dev/null \
       -o ConnectTimeout=10 -o LogLevel=ERROR -o BatchMode=yes \
       -o "ServerAliveInterval=15" -o "ServerAliveCountMax=${alive_count}" \
-      "${ssh_user}@${_GCP_INSTANCE_IP}" "timeout ${timeout} bash -c \"\$(printf '%s' '${encoded_cmd}' | base64 -d)\""
+      "${ssh_user}@${_GCP_INSTANCE_IP}" "timeout ${timeout} bash"
 }
 
 # ---------------------------------------------------------------------------

--- a/sh/e2e/lib/clouds/hetzner.sh
+++ b/sh/e2e/lib/clouds/hetzner.sh
@@ -149,17 +149,15 @@ _hetzner_exec_long() {
   local ip
   ip=$(cat "${ip_file}")
 
-  # Base64-encode the command to avoid shell injection via single-quote breakout
-  local encoded_cmd
-  encoded_cmd=$(printf '%s' "${cmd}" | base64 | tr -d '\n')
-
-  ssh -o StrictHostKeyChecking=no \
+  # Pipe the command via stdin to avoid interpolating it into the remote
+  # command string — eliminates shell injection risk from base64 encoding.
+  printf '%s' "${cmd}" | ssh -o StrictHostKeyChecking=no \
       -o UserKnownHostsFile=/dev/null \
       -o LogLevel=ERROR \
       -o BatchMode=yes \
       -o ConnectTimeout=10 \
       -o ServerAliveInterval=15 \
-      "root@${ip}" "timeout ${timeout_secs} bash -c \"\$(printf '%s' '${encoded_cmd}' | base64 -d)\""
+      "root@${ip}" "timeout ${timeout_secs} bash"
 }
 
 # ---------------------------------------------------------------------------

--- a/sh/e2e/lib/clouds/sprite.sh
+++ b/sh/e2e/lib/clouds/sprite.sh
@@ -216,14 +216,12 @@ _sprite_exec_long() {
   local _max=3
   local _stderr_tmp="/tmp/sprite-execl-err.$$"
 
-  # Base64-encode the command to avoid shell injection via single-quote breakout
-  local encoded_cmd
-  encoded_cmd=$(printf '%s' "${cmd}" | base64 | tr -d '\n')
-
   while [ "${_attempt}" -lt "${_max}" ]; do
     _sprite_fix_config
+    # Pipe the command via stdin to avoid interpolating it into the remote
+    # command string — eliminates shell injection risk from base64 encoding.
     # shellcheck disable=SC2046
-    sprite $(_sprite_org_flags) exec -s "${app}" -- bash -c "timeout '${timeout}' bash -c \"\$(printf '%s' '${encoded_cmd}' | base64 -d)\"" 2>"${_stderr_tmp}"
+    printf '%s' "${cmd}" | sprite $(_sprite_org_flags) exec -s "${app}" -- timeout "${timeout}" bash 2>"${_stderr_tmp}"
     local _rc=$?
     if [ "${_rc}" -eq 0 ]; then
       rm -f "${_stderr_tmp}"

